README.md

```diff
@@ -0,0 +1,73 @@
+# Key Broker Server for SEV
+
+A Key Broker Server (KBS) or Attestation Server verifies the launch measurement of a secure guest and conditionally provides secrets.
+
+This KBS currently supports SEV(-ES) with pre-attestation. 
+The KBS has two gRPC endpoints, `GetBundle` and `GetSecret`. 
+These endpoints are described below and defined in the [protobuf](srcr/grpc/keybroker.proto).
+
+To validate confidential guests, the guest owner must pre-provision the KBS
+with policy information. This KBS uses a database to store policies and secrets.
+The database is described below.
+
+## API
+
+### GetBundle
+The `GetBundle` endpoint taks a policy and certificate chain.
+After verifying the certificate chain, the KBS provides a session file and guest owner public key.
+This session file defines a secure channel between the KBS and the PSP, which will be used to securely provide secrets.
+
+### GetSecret
+The `GetSecret` endpoint takes the launch measurement of the guest, the expected launch parameters, and a secret request. 
+The expected launch parameters such as policy and firmware version are used to select the appropriate owner-approved launch digest. 
+The cryptographically verified launch measurement guarantees the correctness of the launch parameters. 
+
+The secret request allows the user to request multiple secrets in different formats. 
+With an offline KBC, secrets are injected ahead of time. Thus, requesting multiple secrets at once is essential.
```

Comment:
I must confess that I do not understand the "thus" part. When you write "ahead of time", ahead of what exactly, and why is it specific to offline KBCs? And I still don't see why this requires requesting multiple secrets at once (I understand why you need multiple secrets, just not why you could not send multiple requests to get each of them).

Reply:
What it really comes down to is that with SEV(-ES) you can really only inject a secret once. The secret must be formulated inside the trusted domain (i.e. the KBS) and there is no way to join together multiple secret blobs. So any secrets that you need at runtime will need to be in the secret blob generated by the KBS during the `LAUNCH_SECRET` phase of the boot. I guess you could split the secret request into two endpoints, one for telling the KBS which secrets you need that could be called multiple times, and one to generate a secret payload that would include all of those secrets, but I don't think this is any simpler.